prog/mutation.go

```diff
@@ -148,17 +149,128 @@ func (ctx *mutator) removeCall() bool {
 	return true
 }
 
+// Generates a random argument index based on prorities associated with each argument.
+func randomChoice(m map[float64][]int, r *randGen) (int){
+	var keys []float64
+    for k := range m {
+        keys = append(keys, k)
+    }
+
+    // sort the priorities in ascending order
+    sort.Float64s(keys)
+
+    // generate a random value
+	randVal := r.Float64()
+
+	// select the bucket based on the generated value
+	limit := float64(0)
+	for _, p := range(keys) {
+		limit += p * float64(len(m[p])) 
+		if randVal < limit {
+			return m[p][r.rand(len(m[p]))] // uniform random choice
+		}
+	}
+
+	return -1
+}
+
+// Assigns a priority to an argument based on its characteristics.
+func assignArgPriority(i interface{}, r *randGen) float64{
+	maxPrio := float64(100)
+	maxFlags := float64(30) // TODO: is there a limit for the maximum number of flags for a parameter?
+	maxFields := 20
+
+	switch c := i.(type) {
+	case *ArrayType:
```

Comment:
It seems this switch duplicates all cases present in mutationArgs.collectArg.
I think the code will be less fragile and will have more useful locality if we assign priorities right in the mutationArgs.collectArg function. Namely, currently it collects ma.args and ma.ctxes arrays, it could also collect a parallel ma.prios array. Then we don't need this parallel switch.